packages/core/src/TextSplitter.ts

```diff
@@ -18,28 +19,27 @@ class TextSplit {
 
 type SplitRep = { text: string; numTokens: number };
 
-/**
- * Tokenizes sentences. Suitable for English and most European languages.
- * @param text
- * @returns
- */
-export const englishSentenceTokenizer = (text: string) => {
-  // The first part is a lazy match for any character.
-  return text.match(/.+?[.?!]+[\])'"`’”]*(?:\s|$)|.+/g);
+export const defaultSentenceTokenizer = (text: string): string[] => {
+  return nlp(text).sentences().json().map((sentence: any) => sentence.text);
 };
 
-/**
- * Tokenizes sentences. Suitable for Chinese, Japanese, and Korean.
- * @param text
- * @returns
- */
-export const cjkSentenceTokenizer = (text: string) => {
-  // Accepts english style sentence endings with space and
-  // CJK style sentence endings with no space.
-  return text.match(
-    /.+?[.?!]+[\])'"`’”]*(?:\s|$)|.+?[。？！]+[\])'"`’”]*(?:\s|$)?|.+/g,
-  );
-};
+// Refs: https://github.com/fxsjy/jieba/issues/575#issuecomment-359637511
+const resentencesp = /([﹒﹔﹖﹗．；。！？]["’”」』]{0,2}|：(?=["‘“「『]{1,2}|$))/;
```

Comment:
Interesting find. Are you sure this one doesn't have the same DoS issue?

Reply:
<img width="793" alt="image" src="https://github.com/run-llama/LlamaIndexTS/assets/14026360/0422a7ce-ee5d-4fd2-b29b-e2418c86bb97">

https://devina.io/redos-checker